hikari/impl/cache.py

```diff
@@ -142,21 +142,33 @@ def clear(self) -> None:
         self._create_cache()
 
     def clear_dm_channel_ids(self) -> cache.CacheView[snowflakes.Snowflake, snowflakes.Snowflake]:
+        if not self._is_cache_enabled_for(config.CacheComponents.DM_CHANNEL_IDS):
+            return cache_utility.EmptyCacheView()
+
         result = self._dm_channel_entries
-        self._dm_channel_entries = collections.FreezableDict()
+        self._dm_channel_entries = collections.LimitedCapacityCacheMap(limit=self._settings.max_dm_channel_ids)
         return cache_utility.CacheMappingView(result)
 
     def delete_dm_channel_id(
         self, user: snowflakes.SnowflakeishOr[users.PartialUser], /
     ) -> typing.Optional[snowflakes.Snowflake]:
+        if not self._is_cache_enabled_for(config.CacheComponents.DM_CHANNEL_IDS):
+            return None
+
         return self._dm_channel_entries.pop(snowflakes.Snowflake(user), None)
 
     def get_dm_channel_id(
         self, user: snowflakes.SnowflakeishOr[users.PartialUser], /
     ) -> typing.Optional[snowflakes.Snowflake]:
+        if not self._is_cache_enabled_for(config.CacheComponents.DM_CHANNEL_IDS):
+            return None
+
         return self._dm_channel_entries.get(snowflakes.Snowflake(user))
 
     def get_dm_channel_ids_view(self) -> cache.CacheView[snowflakes.Snowflake, snowflakes.Snowflake]:
+        if not self._is_cache_enabled_for(config.CacheComponents.DM_CHANNEL_IDS):
+            return cache_utility.EmptyCacheView()
```

Comment:
test coverage for these?

Reply:
This has been moved down with all the other tests cases for disabled components